src/codegen/compile.ml

```diff
@@ -4529,39 +4603,18 @@ module Serialization = struct
     let name = "@serialize_go<" ^ typ_hash t ^ ">" in
```

Comment:
I think the name of this serialization function will need to depend on the Strm somehow, otherwise you will share code from different Streams, which will be disastrous.

Reply:
See the caveat in the description :-)

Reply:
Oops.